app/models/order_cycle.rb

```diff
@@ -33,6 +36,8 @@ class OrderCycle < ApplicationRecord
   after_save :sync_subscriptions, if: :opening?
 
   validates :name, :coordinator_id, presence: true
+  validate :at_least_one_shipping_method_selected_for_each_distributor
+  validate :no_invalid_order_cycle_shipping_methods
```

Comment:
This may be a problem. I might be wrong but I can imagine that you can switch a shipping method to backend only and suddenly you can't change the same of the order cycle. I know that you are trying to guard against that but sometimes there are bugs or new ways of ding things... I find these validation problems really hard in Rails. Is that a problem? Do you have any good ideas? We could call the validation from the controller only when needed instead of calling it automatically.

We have a similar problem on enterprises which have a `ready_for_checkout` scope and method. So we say that the data is valid but the setup is not ready for checkout. The shop appears closed when it doesn't have a shipping method. And the admin interface let's you know.

This may be a question for the product circle.

Reply:
I thought about this a bit more and I think it would be better to go with a `ready_for_checkout` assessment for order cycles instead of enforcing these validations. At the moment, you can create an order cycle even if your enterprise doesn't have a shipping method. The shop won't show as open because of that. Because of this existing logic I reckon that it's best to see order cycles as not ready for checkout if they don't have valid shipping methods. This can be checked for the selected distributor in the shopfront. Some shops may work then while others don't even though they are in the same order cycle. I think that's fine. If one distributor deletes their shipping method, we don't want to invalidate the order cycle for all other distributors. Does this make sense?

Reply:
Yes that makes sense, I have adjusted things now in https://github.com/openfoodfoundation/openfoodnetwork/pull/9262/commits/6017f07ec84c3c5eb849c1d735d3d52aa60ad295 and https://github.com/openfoodfoundation/openfoodnetwork/pull/9262/commits/5beca7f13ded129e872a27c8637a64da096e8b52